src/rules/lib/parseRuleDocs.ts

```diff
@@ -0,0 +1,46 @@
+import { ImpactDomain, Rule, ScopeName } from '../../types';
+import { promises as fs } from 'fs';
+import { URL } from 'url';
+import yaml from 'js-yaml';
+import { ValidationError } from '../../errors';
+
+interface RulePropertiesSet {
+  rule: string;
+  name: string;
+  title: string;
+  labels?: string[];
+  scope?: ScopeName;
+  impactDomain?: ImpactDomain;
+  references?: Record<string, string>;
+}
+
+export default async function parseRuleDocs(id: string): Promise<Rule> {
+  const content = await fs.readFile(`doc/rules/${id}.md`, 'utf-8');
+  const propertiesContent = content.match(/---\n((?:.*\n)+)---\n((?:.*\n)+?)###/);
+
+  if (!propertiesContent) {
+    throw new ValidationError(`Unable to read properties for rule "${id}".`);
+  }
+
+  const properties = yaml.load(propertiesContent[1]) as RulePropertiesSet;
+
+  return {
+    id: properties.rule,
+    title: properties.title,
+    impactDomain: properties.impactDomain,
+    scope: properties.scope,
```

Comment:
You don’t need to parse the front matter, this metadata is all available on the Rule type.

Reply:
So when building the finding, and desired metadata can be transferred from the Rule - if it’s not already in the findings file?

Reply:
You’ll find scope, impact domain, references etc is already on the Check object

See Check section in the README

https://github.com/applandinc/scanner